### PR TITLE
Trigger validation checks when a new condition is added

### PR DESCRIPTION
**Pull request by kuries**
Parent: #54 
Any changes to PRs #36 and #39 will be available here

### CODE REVIEW

File: backend/app/matrix/background_validator.py [hunk 1]
```diff
@@ -41,66 +47,119 @@ def __init__(
...
+        is_permitted = is_permitted_through_github
+        print(is_permitted, user_id)
+
+        next_state = await self.decide_next_state_of_user(current_state, is_permitted)
+
+        if current_state == next_state:
+            return
+
+        await self.send_next_state_event(room_id, user_id, next_state)
+        await asyncio.sleep(1)
```
Comment (Cadair) [inline backend/app/matrix/background_validator.py:150]: Why do we need this here?

**Review by Cadair — APPROVED** [review]
The only thing I would really like to see changed here is the config option for the background loop.

File: backend/app/matrix/background_validator.py (hunk 1 above)
Comment (kuries) [reply backend/app/matrix/background_validator.py]: That was a copy-paste error of mine :p

File: backend/app/matrix/background_validator.py [hunk 2]
```diff
@@ -41,66 +47,119 @@ def __init__(
         self.github_id_to_matrix_id = dict()
 
         # List of rooms registered under the application and has "room membership" conditions.
-        self.rooms_with_conditions: List[str] = []
+        self.rooms_with_conditions: Dict[str, RegisteredRoom] = dict()
 
         # Set of rooms where the bot has invite and kick permissions.
-        self.rooms_with_permissions: Set[str] = {}
+        self.rooms_with_mod_permissions: Set[str] = {}
```
[the remark this answers is not in the document]
Comment (kuries) [reply backend/app/matrix/background_validator.py]: Thanks!

**Comment by kuries** [comment]
I forgot to handle the part where the updated room and user ids are added into the validator queue. The last commit has those changes.
@Cadair Can you look at that part too, before we merge this PR into PR #54?

**Comment by kuries** [comment]
Merging this PR to include the new changes of the background validator task in the gsoc-final branch.

**Pull request merged**